src/models/tools/table/table-content.ts

```diff
@@ -334,9 +335,21 @@ export const TableContentModel = types
           const parser = new Parser();
           const parsedExpression = parser.parse(expression);
           for (let i = 0; i < attr.values.length; i++) {
-            const xVal = xAttr.value(i) as number;
-            const expressionVal = parsedExpression.evaluate({[kSerializedXKey]: xVal});
-            attr.setValue(i, isFinite(expressionVal) ? expressionVal : "");
+            const xVal = xAttr.value(i) as number | string;
+            if (xVal == null || xVal === "") {
+              attr.setValue(i, undefined);
+            } else {
+              const expressionVal = parsedExpression.evaluate({[kSerializedXKey]: xVal});
+              attr.setValue(i, isFinite(expressionVal) ? expressionVal : NaN);
+            }
+          }
+        } else {
+          for (let i = 0; i < attr.values.length; i++) {
+            const val = attr.value(i);
+            // Clean up displayed errors when an expression is deleted
+            if (Number.isNaN(val as number)) {
```

Comment:
```suggestion
            if (Number.isNaN(val as any)) {
```
for same reason as given above.

Reply:
Dang if I had waited I could have batched the changes...I think there's a third one too